wasm_src/ast_wrapper/ast_wrapper.cc

```diff
@@ -80,129 +80,47 @@ EMSCRIPTEN_KEEPALIVE AstFrameSet* initFrame(const char* header)
+    if (!frameSet || !astIsAFrameSet(frameSet))
     {
-        cout << "Create empty spectral frame failed." << endl;
-        astClearStatus;
+        cout << "Creating frame set failed." << endl;
         return nullptr;
     }
 
-    // set spectral system, unit from parameter (CTYPE, CUNIT)
-    char buffer[128];
-    snprintf(buffer, sizeof(buffer), "System=%s", system);
-    astSet(specframe, buffer);
-    snprintf(buffer, sizeof(buffer), "Unit=%s", unit);
-    astSet(specframe, buffer);
-
-    // get Epoch, ObsLon, ObsLat, ObsAlt, RefRA, RefDec, RestFreq, StdOfRest attributes from AST compound frame
-    // due to observation coordinate (ObsLon/ObsLat/ObsAlt) has to be geodetic, RefRA/RefDec has to be FK5, J2000.
-    // RefRA/RefDec will be 0 if skyframe is not provided, & the conversion will be imprecise.
-    const char *epoch = astGetC(frame, "Epoch");
-    if (epoch)
-    {
-        snprintf(buffer, sizeof(buffer), "Epoch=%s", epoch);
-        astSet(specframe, buffer);
-    }
-    const char *obsLon = astGetC(frame, "ObsLon");
-    if (obsLon)
-    {
-        snprintf(buffer, sizeof(buffer), "ObsLon=%s", obsLon);
-        astSet(specframe, buffer);
-    }
-    const char *obsLat = astGetC(frame, "ObsLat");
-    if (obsLat)
-    {
-        snprintf(buffer, sizeof(buffer), "ObsLat=%s", obsLat);
-        astSet(specframe, buffer);
-    }
-    const char *obsAlt = astGetC(frame, "ObsAlt");
-    if (obsAlt)
-    {
-        snprintf(buffer, sizeof(buffer), "ObsAlt=%s", obsAlt);
-        astSet(specframe, buffer);
-    }
-    const char *refRA = astGetC(frame, "RefRA");
-    if (refRA)
+    // Find spectral frame with spectral template
+    AstSpecFrame *spectralTemplate = astSpecFrame("MaxAxes=100,MinAxes=0");
+    if (!spectralTemplate)
     {
-        snprintf(buffer, sizeof(buffer), "RefRA=%s", refRA);
-        astSet(specframe, buffer);
-    }
-    const char *refDec = astGetC(frame, "RefDec");
-    if (refDec)
-    {
-        snprintf(buffer, sizeof(buffer), "RefDec=%s", refDec);
-        astSet(specframe, buffer);
-    }
-    const char *restFreq = astGetC(frame, "RestFreq");
-    if (restFreq)
-    {
-        snprintf(buffer, sizeof(buffer), "RestFreq=%s", restFreq);
-        astSet(specframe, buffer);
-    }
-    const char *stdOfRest = astGetC(frame, "StdOfRest");
-    if (stdOfRest)
-    {
-        snprintf(buffer, sizeof(buffer), "StdOfRest=%s", stdOfRest);
-        astSet(specframe, buffer);
+        cout << "Creating spectral template failed." << endl;
+        return nullptr;
     }
-
-    if (!astOK)
+    AstFrameSet* found = static_cast<AstFrameSet*>astFindFrame(frameSet, spectralTemplate, " " );
+    if (!found)
     {
-        cout << "Some AST LIB error, check logs." << endl;
-        astClearStatus;
+        cout << "Spectral frame not found." << endl;
         return nullptr;
     }
-    else if (specframe == AST__NULL)
+    AstSpecFrame *specframe = static_cast<AstSpecFrame*>astGetFrame(found, AST__CURRENT);
```

Comment:
Why don't we rather load all the headers into a single frame set, and _then_ split it out into a spectral and spatial frame?

Reply:
For example, `initFrame` could init the entire `AstFrameSet`, and then `getSpatialFrame` and `getSpectralFrame` could pull out the appropriate frames